framework/src/modules/nft/internal_method.ts

```diff
@@ -82,20 +106,95 @@ export class InternalMethod extends BaseMethod {
 		});
 	}
 
-	public async transferInternal(
+	public async verifyTransfer(
+		immutableMethodContext: ImmutableMethodContext,
+		senderAddress: Buffer,
+		nftID: Buffer,
+	) {
+		const owner = await this._nftMethod.getNFTOwner(immutableMethodContext, nftID);
+
+		if (owner.length === LENGTH_CHAIN_ID) {
+			throw new TransferVerifyError(
+				'NFT is escrowed to another chain',
+				NftEventResult.RESULT_NFT_ESCROWED,
+			);
+		}
+
+		if (!owner.equals(senderAddress)) {
+			throw new TransferVerifyError(
+				'Transfer not initiated by the NFT owner',
+				NftEventResult.RESULT_INITIATED_BY_NONOWNER,
+			);
+		}
+
+		const lockingModule = await this._nftMethod.getLockingModule(immutableMethodContext, nftID);
+
+		if (lockingModule !== NFT_NOT_LOCKED) {
+			throw new TransferVerifyError(
+				'Locked NFTs cannot be transferred',
+				NftEventResult.RESULT_NFT_LOCKED,
+			);
+		}
+	}
+
+	public async verifyTransferCrossChain(
```

Comment:
Similar comment as above, please see the ordering of checks and functions used in LIP [here](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0052.md?plain=1#L1382-L1426).

@bobanm Nice catch to optimize the transfer verification checks in the common part and the cross-chain specific part, in order to avoid repetition. Hope the auditors won't mind the difference